secrets/vault/vault.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2019 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limtations under the License.
+
+// Package vault provides functionality to access and decrypt secrets using
+// Hashicorp Vault transit API.
+package vault
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"path"
+
+	"github.com/hashicorp/vault/api"
+	"gocloud.dev/secrets"
+)
+
+// Config is the authentication configurations of the Vault server.
```

Comment:
Consider dropping this type and just putting the two values into `Dial` args.

Document the args (e.g., it looks like `Token` is optional?). Add a link to the public documentation.

Reply:
The value of combining them together is that it may grow in future. The vault's api.Config includes only a subset of the configurations that api.Client can set. We may need to add them in future so this provides the flexibility. I've added some comments talking about the token.